packages/dd-trace/test/setup/tap.js

```diff
@@ -1,4 +1,21 @@
 'use strict'
 
-require('tap').mochaGlobals()
+require('tap')
+const { describe, it, before, beforeEach, after, afterEach } = require('@tapjs/mocha-globals')
+globalThis.describe = describe
+globalThis.it = it
+globalThis.before = patchHook(before)
+globalThis.after = patchHook(after)
+globalThis.beforeEach = patchHook(beforeEach)
+globalThis.afterEach = patchHook(afterEach)
 require('./core')
+
+function promisify(fn) {
+  return () => new Promise(done => fn(done))
+}
+
+function patchHook(hook) {
+  return function patchedHook(fn) {
+    return hook(fn.length ? promisify(fn) : fn)
+  }
+}
```

Comment:
You should be able to replace these lines with this:

```suggestion
require('@tapjs/mocha-globals').globalize()
require('./core')
```

Reply:
Unless there's been a change since I put this together, this was necessary as the hook functions only accepted promise-returning functions and not callbacks, but a bunch of our tests use callbacks in hook functions presently.